src/app/component/ProductCustomizableOptions/ProductCustomizableOptions.container.js

```diff
@@ -0,0 +1,222 @@
+ * See LICENSE for license details.
+ *
+ * @license OSL-3.0 (Open Software License ("OSL") v. 3.0)
+ * @package scandipwa/base-theme
+ * @link https://github.com/scandipwa/base-theme
+ */
+
+import { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { formatCurrency } from 'Util/Price';
+import ProductCustomizableOptions from './ProductCustomizableOptions.component';
+
+export class ProductCustomizableOptionsContainer extends PureComponent {
+    static propTypes = {
+        options: PropTypes.array,
+        getRequiredCustomizableOptions: PropTypes.func.isRequired,
+        getSelectedCustomizableOptions: PropTypes.func.isRequired
+    };
+
+    static defaultProps = {
+        options: []
+    };
+
+    state = {
+        isLoading: true,
+        selectedCheckboxValues: [],
+        selectedDropdownOptions: [],
+        selectedFieldValue: [],
+        selectedAreaValue: []
+    };
+
+    containerFunctions = {
+        setCustomizableOptionFieldValue: this.setCustomizableOptionFieldValue.bind(this),
+        setCustomizableOptionAreaValue: this.setCustomizableOptionAreaValue.bind(this),
+        setSelectedDropdownValue: this.setSelectedDropdownValue.bind(this),
+        getDropdownOptions: this.getDropdownOptions.bind(this),
+        setSelectedCheckboxValues: this.setSelectedCheckboxValues.bind(this)
+    };
+
+    componentDidUpdate(prevProps, prevState) {
+        const { options } = this.props;
+        const { options: prevOptions } = prevProps;
+        const {
+            selectedCheckboxValues,
+            selectedDropdownOptions,
+            selectedFieldValue,
+            selectedAreaValue
+        } = this.state;
+        const {
+            selectedCheckboxValues: prevSelectedCheckboxValues,
+            selectedDropdownOptions: prevSelectedDropdownOptions,
+            selectedFieldValue: prevFieldValue,
+            selectedAreaValue: prevAreaValue
+        } = prevState;
+
+        if (options) {
+            this.stopLoading();
+        }
+
+        if (options !== prevOptions) {
+            this.getRequiredOptionsData(options);
+        }
+
+        if (selectedCheckboxValues !== prevSelectedCheckboxValues) {
+            this.updateSelectedOptionsArray();
+        }
+
+        if (selectedFieldValue !== prevFieldValue
+            || selectedAreaValue !== prevAreaValue
+            || selectedDropdownOptions !== prevSelectedDropdownOptions
+        ) {
+            this.updateSelectedOptions();
+        }
+    }
+
+    stopLoading() {
+        this.setState({ isLoading: false });
+    }
+
+    async updateSelectedOptionsArray() {
+        const { getSelectedCustomizableOptions } = this.props;
+        const { selectedCheckboxValues } = this.state;
+        const customizable_options = [];
+
+        if (selectedCheckboxValues.length) {
+            selectedCheckboxValues.map(item => customizable_options.push(item));
```

Comment:
The use of `map` here is redundant, prefer `forEach`. And in this case prefer the array destructing, i.e.:

```js
customizable_options.push(...selectedCheckboxValues);
```

And I also believe check for length is redundant.